Fix surveys exports with free text in multiple option

Quentinchampenois opened the pull request:
#### :tophat: What? Why?

When survey contains answer with multiple answer choices, exports can be difficult to read for administrators. So this PR aims to change format of answer choices in exports. The format is now : 

* When free text is disabled or user just check option without filling free text
`"answer choice"`

* When free text is enabled and user fill free text
`"answer choice (free text)"`

* When there is several choices
`"answer choice 1 (free text 1), answer choice 2 (free text 2)"` or `"answer choice 1 (free text 1), answer choice 2"`

Also, admin response show view can be enhanced when displaying multiple choice answers.

#### :pushpin: Related Issues
- Fixes #6087

#### Testing

**Test format in exports**
1. Create a survey with sorting and multiple choices questions (enabling free text)
2. Answer the survey
3. Export the answers in admin panel
4. See answer choices as Array in exports

**Test admin response show view**
1. Create a survey with sorting and multiple choices questions (enabling free text)
2. Answer the survey
3. Access responses index
4. Click on a response eye icon
5. See display updated

### :camera: Screenshots
**Admin response show view**
<img width="1037" alt="Screenshot 2021-12-20 at 16 20 56" src="https://user-images.githubusercontent.com/26109239/146903232-f79f36dd-af8c-41dd-8c0c-40580966dd1f.png">


**Exports**
**CSV**
<img width="1114" alt="Screenshot 2021-12-20 at 16 22 35" src="https://user-images.githubusercontent.com/26109239/146903172-60fdad40-c8e2-439e-a9c4-3c56487d233a.png">


#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](https://github.com/decidim/decidim/blob/develop/CONTRIBUTING.adoc) to this repository.

- [x] :question: **CONSIDER** adding a unit test if your PR resolves an issue.
- [x] :heavy_check_mark: **DO** check open PR's to avoid duplicates.
- [x] :heavy_check_mark: **DO** keep pull requests small so they can be easily reviewed.
- [x] :heavy_check_mark: **DO** build locally before pushing.
- [x] :heavy_check_mark: **DO** make sure tests pass.
- [x] :heavy_check_mark: **DO** make sure any new changes are documented in `docs/`.
- [x] :heavy_check_mark: **DO** add and modify seeds if necessary.
- [x] :heavy_check_mark: **DO** add CHANGELOG upgrade notes if required.
- [x] :heavy_check_mark: **DO** add to GraphQL API if there are new public fields.
- [x] :heavy_check_mark: **DO** add link to MetaDecidim if it's a new feature.
- [x] :x:**AVOID** breaking the continuous integration build.
- [x] :x:**AVOID** making significant changes to the overall architecture.

#### Additional context
I tried to simplify my fix however the forms / surveys are complex to understand

:hearts: Thank you!

andreslucena reviewed (changes requested):
@Quentinchampenois I've left you a [comment](https://github.com/decidim/decidim/pull/8582#issuecomment-993478144) about changing format of the export (forgot to mention you, sorry). Let me know if you have a better idea of format, I think my proposal is a bit cleaner and easier to parse for humans 😄

Quentinchampenois commented:
Hello,

Yes I agree this is more human readable, I will update the format ! I am just wondering how it will render with a long text, but maybe it will be fine.

I convert this PR as draft and open it again once it is ready

Quentinchampenois commented:
Hello @carolromero @andreslucena,

I've refactored the current format for options, now it should be `Jazz,Soul,Other(Rock)` in exports. Also I've updated the PR description with a screenshot of current CSV export and JSON export.

Let me know if you see other needs !

andreslucena commented on decidim-forms/app/presenters/decidim/forms/admin/questionnaire_answer_presenter.rb:
```diff
@@ -52,11 +55,21 @@ def pretty_attachment(attachment)
+        def body_or_custom_body(choice)
+          return choice.custom_body if choice.try(:custom_body).present?
+
+          choice.try(:body).present? ? "-" : ""
```
I don't like this "-" thing, but as far as I see this is already the behavior of `decidim-forms`, so we'll leave it as is

andreslucena commented:
Oh, by seeing the UI I see that it isn't fixed in the "Show responses" page @Quentinchampenois 

What we have now is:

![image](https://user-images.githubusercontent.com/717367/146747780-2021b27c-5d9e-41ac-8eec-6c5706d3dc4a.png)


What it should be is 

* Rock
* Jazz
* Otro (Blues)

For comparison, this is what we have in `develop` now:

![image](https://user-images.githubusercontent.com/717367/146747913-09df3959-b21b-4579-9384-2a8ca9a72314.png)

Can you change it 🙏🏽 ?

Quentinchampenois commented:
Hello @andreslucena,

Thank you for your review, I've updated the UI answer presenter. According to your comment on the questionnaire answer presenter I agree and I also dislike this "-" char. I've decided to remove it and let it empty, it is more human readable.

I update the UI screenshot and CSV screenshot in the description, I hope it is the expected behaviour, tell me if you see other points

andreslucena reviewed (approved):
👍🏽

andreslucena commented:
As this is a fix, could you make a backport to v0.25-stable @Quentinchampenois  🙏🏽?
